plugins/rest/rest_auth.go

```diff
@@ -415,6 +420,7 @@ type awsSigningAuthPlugin struct {
 	AWSEnvironmentCredentials *awsEnvironmentCredentialService `json:"environment_credentials,omitempty"`
 	AWSMetadataCredentials    *awsMetadataCredentialService    `json:"metadata_credentials,omitempty"`
 	AWSWebIdentityCredentials *awsWebIdentityCredentialService `json:"web_identity_credentials,omitempty"`
+	AWSService                *string                          `json:"service,omitempty"`
```

Comment:
Sorry for chasing you around the block for this, but I've just now considered this:

- This setting is new, so backwards-compat is not an issue.
- Having a `*string` type is usually the thing you'd do if you want to **discern** `""` (set to an empty string) from `nil` (unset).
- If we **default** to `s3`, and there's no possibility anyone would ever want to configure this to `""`, we could simplify this to `string`, and not worry about the const/var/pointer business at all.

WDYT? 😃

Reply:
Much neater! Thanks! :)